diff -urN vdr-1.7.21/channels.h vdr-1.7.21b/channels.h
--- vdr-1.7.21/channels.h	2011-08-06 11:56:13.000000000 +0200
+++ vdr-1.7.21b/channels.h	2011-09-10 17:01:09.997012068 +0200
@@ -14,6 +14,7 @@
 #include "sources.h"
 #include "thread.h"
 #include "tools.h"
+#include <vector>
 
 #define ISTRANSPONDER(f1, f2)  (abs((f1) - (f2)) < 4) //XXX
 
@@ -90,6 +91,9 @@
   friend class cSchedules;
   friend class cMenuEditChannel;
   friend class cDvbSourceParam;
+  friend class cMenuMyEditChannel;
+  friend class cMenuMyBouquets;
+  friend class cMenuEditBouquet;
 private:
   static cString ToText(const cChannel *Channel);
   char *name;
@@ -173,6 +177,7 @@
   int Number(void) const { return number; }
   void SetNumber(int Number) { number = Number; }
   bool GroupSep(void) const { return groupSep; }
+  void SetGroupSep(bool GroupSep) { groupSep=GroupSep; }
   const char *Parameters(void) const { return parameters; }
   const cLinkChannels* LinkChannels(void) const { return linkChannels; }
   const cChannel *RefChannel(void) const { return refChannel; }
@@ -207,6 +212,7 @@
 public:
   cChannels(void);
   bool Load(const char *FileName, bool AllowComments = false, bool MustExist = false);
+  bool Reload(const char *FileName, bool AllowComments = false, bool MustExist = false);
   void HashChannel(cChannel *Channel);
   void UnhashChannel(cChannel *Channel);
   int GetNextGroup(int Idx);   // Get next channel group
diff -urN vdr-1.7.21/config.c vdr-1.7.21b/config.c
--- vdr-1.7.21/config.c	2011-08-20 11:12:05.000000000 +0200
+++ vdr-1.7.21b/config.c	2011-09-10 17:02:32.088965761 +0200
@@ -399,6 +399,16 @@
   InitialVolume = -1;
   ChannelsWrap = 0;
   EmergencyExit = 1;
+  ExpertOptions  = 0;
+  AddNewChannels = 0;
+  UseBouquetList = 1;
+  OnlyRadioChannels = 0;
+  OnlyEncryptedChannels = 0;
+  OnlyHDChannels = 0;
+  ExpertNavi     = 0;
+  WantChListOnOk = 1;
+  ChannelUpDownKeyMode = 0; // 0 Normal, kChanUp and kChanDn jumps channel
+                            // 1 opens bouquet / channellist
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -503,6 +513,15 @@
 bool cSetup::Parse(const char *Name, const char *Value)
 {
   if      (!strcasecmp(Name, "OSDLanguage"))       { strn0cpy(OSDLanguage, Value, sizeof(OSDLanguage)); I18nSetLocale(OSDLanguage); }
+  else if (!strcasecmp(Name, "ExpertOptions"))       ExpertOptions      = atoi(Value);
+  else if (!strcasecmp(Name, "AddNewChannels"))      AddNewChannels     = atoi(Value);
+  else if (!strcasecmp(Name, "UseBouquetList"))      UseBouquetList     = atoi(Value);
+  else if (!strcasecmp(Name, "OnlyRadioChannels"))   OnlyRadioChannels  = atoi(Value);
+  else if (!strcasecmp(Name, "OnlyEncryptedChannels"))OnlyEncryptedChannels = atoi(Value);
+  else if (!strcasecmp(Name, "OnlyHDChannels"))      OnlyHDChannels       = atoi(Value);
+  else if (!strcasecmp(Name, "ExpertNavi"))          ExpertNavi         = atoi(Value);
+  else if (!strcasecmp(Name, "WantChListOnOk"))      WantChListOnOk     = atoi(Value);
+  else if (!strcasecmp(Name, "ChannelUpDownKeyMode"))  ChannelUpDownKeyMode = atoi(Value);
   else if (!strcasecmp(Name, "OSDSkin"))             Utf8Strn0Cpy(OSDSkin, Value, MaxSkinName);
   else if (!strcasecmp(Name, "OSDTheme"))            Utf8Strn0Cpy(OSDTheme, Value, MaxThemeName);
   else if (!strcasecmp(Name, "PrimaryDVB"))          PrimaryDVB         = atoi(Value);
@@ -687,6 +706,15 @@
   Store("InitialVolume",      InitialVolume);
   Store("ChannelsWrap",       ChannelsWrap);
   Store("EmergencyExit",      EmergencyExit);
+  Store("ExpertOptions",      ExpertOptions);
+  Store("AddNewChannels",     AddNewChannels);
+  Store("UseBouquetList",     UseBouquetList);
+  Store("OnlyRadioChannels",  OnlyRadioChannels);
+  Store("OnlyEncryptedChannels", OnlyEncryptedChannels);
+  Store("OnlyHDChannels", OnlyHDChannels);
+  Store("ExpertNavi",         ExpertNavi);
+  Store("WantChListOnOk",     WantChListOnOk);
+  Store("ChannelUpDownKeyMode", ChannelUpDownKeyMode);
 
   Sort();
 
diff -urN vdr-1.7.21/config.h vdr-1.7.21b/config.h
--- vdr-1.7.21/config.h	2011-08-20 10:51:47.000000000 +0200
+++ vdr-1.7.21b/config.h	2011-09-10 17:12:01.816961935 +0200
@@ -13,6 +13,7 @@
 #include <arpa/inet.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string>
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
@@ -290,6 +291,16 @@
   int InitialVolume;
   int ChannelsWrap;
   int EmergencyExit;
+  int ExpertOptions;
+  int AddNewChannels;
+  int UseBouquetList;
+  int OnlyRadioChannels;
+  int OnlyEncryptedChannels;
+  int OnlyHDChannels;
+  int ExpertNavi;
+  int WantChListOnOk;
+  int ChannelUpDownKeyMode; // 0 Normal, jumps to next or previous channel
+                            // 1 opens bouquet / channel list
   int __EndData__;
   cString InitialChannel;
   cSetup(void);
diff -urN vdr-1.7.21/debug.h vdr-1.7.21b/debug.h
--- vdr-1.7.21/debug.h	1970-01-01 01:00:00.000000000 +0100
+++ vdr-1.7.21b/debug.h	2011-09-10 16:59:11.188980201 +0200
@@ -0,0 +1,44 @@
+
+#ifndef DEBUG_H
+#define DEBUG_H
+#include <assert.h>
+
+#ifdef DBG
+  #undef DBG
+  #undef DERR
+  #undef DLOG
+  #undef PRINTF
+#endif
+
+
+#if defined DEBUG
+    #ifdef PLUGIN_NAME_I18N
+        #define CONTEXT PLUGIN_NAME_I18N
+    #else
+        #define CONTEXT "VDR"
+    #endif
+
+    // standard debug macro
+    #define DDD(x...) { printf("["CONTEXT": %s +%d, %s()]: ",__FILE__, __LINE__, __FUNCTION__); \
+                         printf(x); printf("\n"); }
+
+    // standard error macro (red background)
+    #define DERR(x...) { printf("\033[0;37m\033[1;41m["CONTEXT"]: "x); printf("\033[0m\n"); }
+
+    // special macro: logs to syslog AND console
+    #define DLOG(x...) { printf("["CONTEXT"]: "x); printf("\n"); dsyslog("["CONTEXT"]: "x); }
+
+    // for automatic conversion of printf's - i.e. use sed s/" printf"/" PRINTF"/
+    #define PRINTF(x...) printf("["CONTEXT"]: "x)
+#else
+    #define DDD(x...)
+    #define DERR(x...)
+    #define DLOG(x...)
+    #define PRINTF(x...)
+#endif
+
+//only here for compatibility reasons, don't use any more
+#define ERR "Error"
+#define DBG(x...)
+
+#endif /* DEBUG_H */
diff -urN vdr-1.7.21/keys.c vdr-1.7.21b/keys.c
--- vdr-1.7.21/keys.c	2010-04-05 12:05:58.000000000 +0200
+++ vdr-1.7.21b/keys.c	2011-09-10 17:04:06.008976614 +0200
@@ -56,6 +56,9 @@
                     { kRecordings,    trNOOP("Key$Recordings")  },
                     { kSetup,         trNOOP("Key$Setup")       },
                     { kCommands,      trNOOP("Key$Commands")    },
+                    { kGreater,       trNOOP("Key$Greater")     },
+                    { kLess,          trNOOP("Key$Less")        },
+                    { k2digit,        trNOOP("Key$2digit")      },
                     { kUser0,         trNOOP("Key$User0")       },
                     { kUser1,         trNOOP("Key$User1")       },
                     { kUser2,         trNOOP("Key$User2")       },
diff -urN vdr-1.7.21/keys.h vdr-1.7.21b/keys.h
--- vdr-1.7.21/keys.h	2010-04-05 12:06:04.000000000 +0200
+++ vdr-1.7.21b/keys.h	2011-09-10 17:04:17.708918546 +0200
@@ -50,6 +50,7 @@
              kRecordings,
              kSetup,
              kCommands,
+             kGreater, kLess, k2digit,
              kUser0, kUser1, kUser2, kUser3, kUser4, kUser5, kUser6, kUser7, kUser8, kUser9,
              kNone,
              kKbd,
diff -urN vdr-1.7.21/menu.c vdr-1.7.21b/menu.c
--- vdr-1.7.21/menu.c	2011-08-27 13:05:33.000000000 +0200
+++ vdr-1.7.21b/menu.c	2011-09-10 17:15:29.656958186 +0200
@@ -131,15 +131,6 @@
 
 // --- cMenuEditSrcItem ------------------------------------------------------
 
-class cMenuEditSrcItem : public cMenuEditIntItem {
-private:
-  const cSource *source;
-protected:
-  virtual void Set(void);
-public:
-  cMenuEditSrcItem(const char *Name, int *Value);
-  eOSState ProcessKey(eKeys Key);
-  };
 
 cMenuEditSrcItem::cMenuEditSrcItem(const char *Name, int *Value)
 :cMenuEditIntItem(Name, Value, 0)
@@ -3768,6 +3759,7 @@
          else
             return osEnd;
          break;
+    case kInfo: return osEnd; break;
     default:
          if ((Key & (k_Repeat | k_Release)) == 0) {
             cRemote::Put(Key);
diff -urN vdr-1.7.21/menu.h vdr-1.7.21b/menu.h
--- vdr-1.7.21/menu.h	2010-03-06 17:15:59.000000000 +0100
+++ vdr-1.7.21b/menu.h	2011-09-10 17:07:21.760969487 +0200
@@ -19,6 +19,18 @@
 #include "recorder.h"
 #include "skins.h"
 
+
+class cMenuEditSrcItem : public cMenuEditIntItem {
+private:
+  const cSource *source;
+protected:
+  virtual void Set(void);
+public:
+  cMenuEditSrcItem(const char *Name, int *Value);
+  eOSState ProcessKey(eKeys Key);
+  };
+
+
 class cMenuText : public cOsdMenu {
 private:
   char *text;
@@ -95,7 +107,24 @@
   virtual eOSState ProcessKey(eKeys Key);
   };
 
+struct MenuMainHook_Data_V1_0 {
+  eOSState  Function;    /*IN*/
+  cOsdMenu *pResultMenu; /*OUT*/
+};
+#define CREATE_MENU_MAIN(function,menu) {                              \
+  MenuMainHook_Data_V1_0 data;                                         \
+  memset(&data, 0, sizeof(data));                                      \
+  data.Function = function;                                            \
+  if (!cPluginManager::CallFirstService("MenuMainHook-V1.0", &data))   \
+    menu = new cMenuMain(function);                                    \
+  else                                                                 \
+    menu = data.pResultMenu;                                           \
+}
+
+
 class cMenuMain : public cOsdMenu {
+protected:
+  cMenuMain(const char *Title):cOsdMenu(Title){}; // Allow custom constructor
 private:
   bool replaying;
   cOsdItem *stopReplayItem;
diff -urN vdr-1.7.21/osdbase.h vdr-1.7.21b/osdbase.h
--- vdr-1.7.21/osdbase.h	2010-01-16 15:25:31.000000000 +0100
+++ vdr-1.7.21b/osdbase.h	2011-09-10 16:59:39.592986163 +0200
@@ -33,6 +33,10 @@
                 osSwitchDvb,
                 osBack,
                 osEnd,
+                osBouquets,
+                osActiveBouquet,
+                osFavourites,
+                osAddFavourite,
                 os_User, // the following values can be used locally
                 osUser1,
                 osUser2,
diff -urN vdr-1.7.21/vdr.c vdr-1.7.21b/vdr.c
--- vdr-1.7.21/vdr.c	2011-08-15 14:42:39.000000000 +0200
+++ vdr-1.7.21b/vdr.c	2011-09-10 17:17:02.928956244 +0200
@@ -1039,6 +1039,28 @@
           case kChanUp:
           case kChanDn|k_Repeat:
           case kChanDn:
+
+               if (Setup.ChannelUpDownKeyMode == 1 && (!Menu || cDisplayChannel::IsOpen()))
+               // only if a menu is not open
+               //  and user has opted for a different key behavior
+               {
+                   if (cDisplayChannel::IsOpen() || cControl::Control())
+                   {
+                       DELETE_MENU;
+                   }
+                   if (key == kChanDn)
+                   {
+                       DirectMainFunction(osBouquets);
+                   }
+                   else
+                   {
+                       DirectMainFunction(osActiveBouquet);
+                   }
+                   key = kNone; // nobody else sees these keys
+                   break;
+               }
+               else
+
                if (!Interact)
                   Menu = new cDisplayChannel(NORMALKEY(key));
                else if (cDisplayChannel::IsOpen() || cControl::Control()) {
@@ -1082,6 +1104,7 @@
                break;
           // Subtitle track control:
           case kSubtitles:
+         case kGreater:
                if (cControl::Control())
                   cControl::Control()->Hide();
                if (!cDisplaySubtitleTracks::IsOpen()) {
@@ -1237,7 +1260,19 @@
                   Menu = new cDisplayChannel(NORMALKEY(key));
                   break;
              // Viewing Control:
-             case kOk:   LastChannel = -1; break; // forces channel display
+             case kInfo:
+             case kOk:  if ((Setup.WantChListOnOk && (key == kOk)) 
+                            || (!Setup.WantChListOnOk && (key == kInfo)) )
+                        {
+                            DirectMainFunction(osChannels);
+                        }
+                        else 
+                        {
+                            LastChannel = -1;
+                        }
+                        break;
+
+
              // Instant resume of the last viewed recording:
              case kPlay:
                   if (cReplayControl::LastReplayed()) {
