Index: device.c
===================================================================
--- device.c	(revision 15385)
+++ device.c	(revision 15386)
@@ -21,6 +21,28 @@
 
 // --- cLiveSubtitle ---------------------------------------------------------
 
+#ifdef REELVDR
+/***
+*  returns True if Tuner-device with priority < 0 is found
+*  priority < 0 implies No recording
+*  live Viewing = -1 priority
+*/
+bool AnyFreeTuners()
+{
+    cDevice* device=NULL;
+    for (int tuner=0; tuner<MAXDEVICES ; tuner++)
+    {
+        device = cDevice::GetDevice(tuner);
+        printf("tuner#%d : priority %d \n",tuner, device?device->Priority():-100 );
+        if ( device &&
+             ( device->ProvidesSource(cSource::stSat) || device->ProvidesSource(cSource::stCable) || device->ProvidesSource(cSource::stTerr) ) && // device is a Tuner
+             device->Priority()<0 ) // free Tuner: ie. with no recordings in this tuner
+            return true;
+    } // end for
+    return false; // no device found
+}
+#endif /* REELVDR */
+
 class cLiveSubtitle : public cReceiver {
 protected:
   virtual void Receive(uchar *Data, int Length);
Index: device.h
===================================================================
--- device.h	(revision 15385)
+++ device.h	(revision 15386)
@@ -25,6 +25,10 @@
 #include "thread.h"
 #include "tools.h"
 
+#ifdef REELVDR
+bool AnyFreeTuners(); // returns true if a tuner without recording is found
+#endif /* REELVDR */
+
 #define MAXDEVICES         16 // the maximum number of devices in the system
 #define MAXPIDHANDLES      64 // the maximum number of different PIDs per device
 #define MAXRECEIVERS       16 // the maximum number of receivers per device
