--- xine-lib-1.1.8/src/input/input_dvd.c	2009-02-20 16:33:01.242528416 +0100
+++ xine-lib-1.1.8.working/src/input/input_dvd.c	2009-02-20 16:33:04.242699386 +0100
@@ -1505,7 +1505,6 @@
     return 0;
   }
 
-  free (locator_orig);
   
   dvdnav_get_title_string(this->dvdnav, &this->dvd_name);
   if(this->dvd_name)
@@ -1554,6 +1553,7 @@
 	      "input_dvd: Title %i is out of range (1 to %i).\n", tt, titles);
       dvdnav_close(this->dvdnav);
       this->dvdnav = NULL;
+      free (locator_orig);
       return 0;
     }
 
@@ -1567,6 +1567,7 @@
 		"input_dvd: Part %i is out of range (1 to %i).\n", pr, parts);
 	dvdnav_close(this->dvdnav);
 	this->dvdnav = NULL;
+	free (locator_orig);
 	return 0;
       }
     }
@@ -1598,7 +1599,8 @@
   xine_event_send(this->stream, &event);
 
   update_title_display(this);
-  
+
+  free (locator_orig);
   return 1;
 }
 
